src/test/java/com/nvidia/spark/rapids/jni/RmmSparkTest.java

```diff
@@ -79,6 +80,71 @@ public void testInsertOOMs() {
     }
   }
 
+  @Test
+  public void testInsertMultipleOOMs() {
+    Rmm.initialize(RmmAllocationMode.CUDA_DEFAULT, null, 512 * 1024 * 1024);
+    RmmSpark.setEventHandler(new BaseRmmEventHandler());
+    long threadId = RmmSpark.getCurrentThreadId();
+    long taskid = 0; // This is arbitrary
+    RmmSpark.associateThreadWithTask(threadId, taskid);
+    try {
+      // Allocate something small and verify that it works...
+      Rmm.alloc(100).close();
+
+      // Force an exception
+      int numRetryOOMs = 3;
+      RmmSpark.forceRetryOOM(threadId, numRetryOOMs);
+      for (int i = 0; i < numRetryOOMs; i++) {
+        assertThrows(RetryOOM.class, () -> Rmm.alloc(100).close());
+        // Verify that injecting OOM does not cause the block to actually happen
+        RmmSpark.blockThreadUntilReady();
+      }
+
+      // Allocate something small and verify that it works...
+      Rmm.alloc(100).close();
+
+      // Force another exception
+      int numSplitAndRetryOOMs = 5;
+      RmmSpark.forceSplitAndRetryOOM(threadId, numSplitAndRetryOOMs);
+      for (int i = 0; i < numSplitAndRetryOOMs; i++) {
+        assertThrows(SplitAndRetryOOM.class, () -> Rmm.alloc(100).close());
+        // Verify that injecting OOM does not cause the block to actually happen
+        RmmSpark.blockThreadUntilReady();
+      }
+
+      // Allocate something small and verify that it works...
+      Rmm.alloc(100).close();
+    } finally {
+      RmmSpark.removeThreadAssociation(threadId);
+    }
+  }
+
+  public void testCudfException() {
```

Comment:
Sorry I just noticed. IS this test intended to be run as a test? Shouldn't it be tagged with `@Test`???

Reply:
yikes, indeed it was not running. It is now though. Nice catch